skia-safe/src/effects/layer_draw_looper.rs

```diff
@@ -0,0 +1,92 @@
+use crate::prelude::*;
+use crate::{BlendMode, DrawLooper, Vector};
+use skia_bindings::{
+    C_SkLayerDrawLooper_Builder_destruct, C_SkLayerDrawLooper_Builder_detach,
+    SkLayerDrawLooper_Builder, SkLayerDrawLooper_LayerInfo, SkPaint,
+};
+
+bitflags! {
+    pub struct LayerDrawLooperBits : u32 {
+        const STYLE = skia_bindings::SkLayerDrawLooper_Bits_kStyle_Bit as _;
+        const PATH_EFFECT = skia_bindings::SkLayerDrawLooper_Bits_kPathEffect_Bit as _;
+        const MASK_FILTER = skia_bindings::SkLayerDrawLooper_Bits_kMaskFilter_Bit as _;
+        const SHADER = skia_bindings::SkLayerDrawLooper_Bits_kShader_Bit as _;
+        const COLOR_FILTER = skia_bindings::SkLayerDrawLooper_Bits_kColorFilter_Bit as _;
+        const XFER_MODE = skia_bindings::SkLayerDrawLooper_Bits_kXfermode_Bit as _;
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum LayerDrawLooperBitFlags {
+    Bits(LayerDrawLooperBits),
+    EntirePaint,
+}
+
+impl From<LayerDrawLooperBits> for LayerDrawLooperBitFlags {
+    fn from(bits: LayerDrawLooperBits) -> Self {
+        LayerDrawLooperBitFlags::Bits(bits)
+    }
+}
+
+#[derive(Clone, PartialEq, Debug)]
+pub struct LayerDrawLooperLayerInfo {
+    pub paint_bits: LayerDrawLooperBitFlags,
+    pub color_mode: BlendMode,
+    pub offset: Vector,
+    pub post_translate: bool,
+}
+
+pub type LayerDrawLooperBuilder = Handle<SkLayerDrawLooper_Builder>;
+
+impl NativeDrop for SkLayerDrawLooper_Builder {
+    fn drop(&mut self) {
+        unsafe { C_SkLayerDrawLooper_Builder_destruct(self) }
+    }
+}
+
+impl Default for LayerDrawLooperBuilder {
+    fn default() -> LayerDrawLooperBuilder {
+        Self::from_native(unsafe { SkLayerDrawLooper_Builder::new() })
+    }
+}
+
+impl From<&LayerDrawLooperLayerInfo> for SkLayerDrawLooper_LayerInfo {
```

Comment:
This should probably be a function in LayerDrawLooperLayerInfo::to_native(), because it does not consume its source.